src/Microsoft.ML.Transforms/WrappedWhiteningTransformer.cs

```diff
@@ -0,0 +1,165 @@
+                input.Schema.TryGetColumnIndex(inputColumn, out int col);
+                typesmapping[inputColumn] = new VectorType(NumberType.R4, input.Schema.GetColumnType(col).VectorSize);
+            }
+
+            var schema = new SimpleSchema(Host, typesmapping.ToArray());
+            var emptyData = new EmptyDataView(Host, schema);
+
+            return new TransformWrapper(Host, new WhiteningTransform(Host, args, emptyData));
+        }
+    }
+
+    /// <summary>
+    /// Extensions for statically typed Whitening estimator.
+    /// </summary>
+    public static class WhiteningExtensions
+    {
+        private sealed class OutPipelineColumn : Vector<float>
+        {
+            public readonly Vector<float> Input;
+
+            public OutPipelineColumn(Vector<float> input, WhiteningKind kind, float eps, int maxRows, bool saveInverse, int pcaNum)
+                : base(new Reconciler(kind, eps, maxRows, saveInverse, pcaNum), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler : EstimatorReconciler
+        {
+            private readonly WhiteningKind _kind;
+            private readonly float _eps;
+            private readonly int _maxRows;
+            private readonly bool _saveInverse;
+            private readonly int _pcaNum;
+
+            public Reconciler(WhiteningKind kind, float eps, int maxRows, bool saveInverse, int pcaNum)
+            {
+                _kind = kind;
+                _eps = eps;
+                _maxRows = maxRows;
+                _saveInverse = saveInverse;
+                _pcaNum = pcaNum;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                Contracts.Assert(toOutput.Length == 1);
+
+                var pairs = new List<(string input, string output)>();
+                foreach (var outCol in toOutput)
+                    pairs.Add((inputNames[((OutPipelineColumn)outCol).Input], outputNames[outCol]));
+
+                return new Whitening(env, pairs.ToArray(), _kind, _eps, _maxRows, _saveInverse, _pcaNum);
+            }
+        }
+
+        /// <include file='doc.xml' path='doc/members/member[@name="Whitening"]/*'/>
+        /// <param name="input">The column to apply to.</param>
+        /// <param name="kind">Whitening kind (PCA/ZCA).</param>
+        /// <param name="eps">Scaling regularizer.</param>
+        /// <param name="maxRows">Max number of rows.</param>
+        /// <param name="saveInverse">Whether to save inverse (recovery) matrix.</param>
+        /// <param name="pcaNum">PCA components to retain.</param>
+        public static Vector<float> Whiten(this Vector<float> input,
```

Comment:
>Whiten [](start = 36, length = 6)

this sounds like a wacky thing to do to an arbitrary float array, is it not?
Do you think there is a good name that's more elaborate?

And if not, maybe consider not having the pigstension at all?